Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -1982,10 +1116,16 @@ void LoadEventNexus::loadEvents(API::Progress *const prog,
 
   for (size_t i = bank0; i < bankn; i++) {
     // We make tasks for loading
-    if (bankNumEvents[i] > 0)
-      pool.schedule(new LoadBankFromDiskTask(
+    if (bankNumEvents[i] > 0) {
+      LoadBankFromDiskTask *task = new LoadBankFromDiskTask(
           this, bankNames[i], classType, bankNumEvents[i], oldNeXusFileNames,
-          prog2, diskIOMutex, scheduler, periodLogVec));
+          prog2, diskIOMutex, scheduler, periodLogVec, this->getLogger());
+      task->run();
+
```

Comment:
There is now a memory leak as `task` is not deleted. If this structure is kept then we should use a `Mantid::Kernel::make_unique` and a `std::unique_ptr` to manage the memory.